server/src/test/java/org/opensearch/telemetry/tracing/SpanBuilderTests.java

```diff
@@ -36,20 +36,22 @@ public void testHttpRequestContext() {
         HttpRequest httpRequest = createHttpRequest();
         SpanCreationContext context = SpanBuilder.from(httpRequest);
         Attributes attributes = context.getAttributes();
-        assertEquals("GET /_test", context.getSpanName());
+        assertEquals("GET /_test/resource", context.getSpanName());
```

Comment:
The way test is modified right now - it only checks URIs with query strings, I would suggest to convert it to parameterized test case and exercise different combinations